debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Replicator.java

```diff
@@ -467,10 +472,22 @@ protected void readOplog(MongoClient primary) {
         // Read as much of the oplog as we can ...
         try (MongoCursor<Document> cursor = results.iterator()) {
             while (running.get() && cursor.hasNext()) {
-                if (!handleOplogEvent(primaryAddress, cursor.next())) {
+                final Document event = cursor.next();
+                if (!handleOplogEvent(primaryAddress, event)) {
                     // Something happened, and we're supposed to stop reading
                     return;
                 }
+                try {
+                    heartbeat.heartbeat(source.partition(replicaSet.replicaSetName()), () -> {
+                            source.opLogEvent(replicaSet.replicaSetName(), event);
```

Comment:
Why is `opLogEvent()` called here?

Reply:
@gunnarmorling It moves offsets forward even for message that would be otherwise filtered out - so for MOngoDB hertbeat messages and for non-whitlisted collection changes

Reply:
Ok, got it. I have to say I'm not super happy about the imposed complexity, but I cannot think of a better way right now. What put me off was the fact that `opLogEvent()` would be called twice if a heartbeat happens after a non-filtered event.